src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/ComputeFeatureNeighborMisorientations.cpp

```diff
@@ -0,0 +1,115 @@
+ComputeFeatureNeighborMisorientations::~ComputeFeatureNeighborMisorientations() noexcept = default;
+
+// -----------------------------------------------------------------------------
+const std::atomic_bool& ComputeFeatureNeighborMisorientations::getCancel()
+{
+  return m_ShouldCancel;
+}
+
+// -----------------------------------------------------------------------------
+Result<> ComputeFeatureNeighborMisorientations::operator()()
+{
+
+  std::vector<LaueOps::Pointer> orientationOps = LaueOps::GetAllOrientationOps();
+
+  // Input Arrays
+  const auto& inFeaturePhases = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeaturePhasesArrayPath);
+  const auto& inAvgQuats = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->AvgQuatsArrayPath);
+  const auto& inXtalStruct = m_DataStructure.getDataRefAs<UInt32Array>(m_InputValues->CrystalStructuresArrayPath);
+  const auto& inNeighborList = m_DataStructure.getDataRefAs<NeighborList<int32>>(m_InputValues->NeighborListArrayPath);
+
+  // The output misorientations is going to be used as a vector because the output array is optional and might
+  // not exist in the DataStructure. We cannot get it by reference.
+  auto* avgMisorientations = m_DataStructure.getDataAs<Float32Array>(m_InputValues->AvgMisorientationsArrayName);
+
+  size_t tempMisoList = 0;
+
+  size_t totalFeatures = inFeaturePhases.getNumberOfTuples();
+
+  std::vector<std::vector<float>> tempMisorientationLists(totalFeatures);
+  usize quatIndex = 0;
+  for(size_t i = 1; i < totalFeatures; i++)
+  {
+    quatIndex = i * 4;
+
+    QuatF q1(inAvgQuats[quatIndex], inAvgQuats[quatIndex + 1], inAvgQuats[quatIndex + 2], inAvgQuats[quatIndex + 3]);
+    uint32_t xtalType1 = inXtalStruct[inFeaturePhases[i]];
+
+    const NeighborList<int32_t>::VectorType& featureNeighborList = inNeighborList.getListReference(static_cast<int32_t>(i));
+
+    tempMisorientationLists[i].assign(featureNeighborList.size(), -1.0);
+
+    for(size_t j = 0; j < featureNeighborList.size(); j++)
+    {
+      int32_t neighborFeatureId = featureNeighborList[j];
+      quatIndex = neighborFeatureId * 4;
+      QuatF q2(inAvgQuats[quatIndex], inAvgQuats[quatIndex + 1], inAvgQuats[quatIndex + 2], inAvgQuats[quatIndex + 3]);
+      uint32_t xtalType2 = inXtalStruct[inFeaturePhases[neighborFeatureId]];
+      tempMisoList = featureNeighborList.size();
+      if(xtalType1 == xtalType2 && static_cast<int64_t>(xtalType1) < static_cast<int64_t>(orientationOps.size()))
+      {
+        OrientationD axisAngle = orientationOps[xtalType1]->calculateMisorientation(q1, q2);
+
+        tempMisorientationLists[i][j] = static_cast<float>(axisAngle[3] * nx::core::Constants::k_180OverPiF);
+        if(m_InputValues->ComputeAvgMisors)
+        {
+          (*avgMisorientations)[i] += tempMisorientationLists[i][j];
+        }
+      }
+      else
+      {
+        if(m_InputValues->ComputeAvgMisors)
+        {
+          tempMisoList--;
```

Comment:
```suggestion
          tempMisoList--;
```
I am not sure if this is even possible, but if `featureNeighborList`, aka `inNeighborList` at `i` index, is empty / uninitialized there's a good chance we underflow the number here. Which would do weird things when it's used later

Reply:
Added a check to ensure we don't underflow.